crates/turborepo-lib/src/run/scope/target_selector.rs

```diff
@@ -195,8 +195,12 @@ pub fn is_selector_by_location(
             .iter()
             .any(|prefix| raw_selector.starts_with(prefix))
     {
+        let cleaned_selector = path_clean::clean(std::path::Path::new(raw_selector))
+            .into_os_string()
+            .into_string()
```

Comment:
We could use `AnchoredSystemPathBuf::from_system_path` to avoid these conversion, but that codepath will failif the path starts with `../` which is a valid filter.